src/volatile_memory.rs

```diff
@@ -296,10 +292,18 @@ impl<'a> VolatileSlice<'a> {
     ///
     /// ```
     /// # use vm_memory::VolatileMemory;
-    /// let mut mem = [0u8; 32];
-    /// let mem_ref = &mut mem[..];
-    /// let vslice = mem_ref.get_slice(0, 32).unwrap();
-    /// let (start, end) = vslice.split_at(8).unwrap();
+    /// #
+    /// # // Create a buffer
+    /// # let mut mem = [0u8; 32];
+    /// # let mem_ref = &mut mem[..];
+    /// #
+    /// # // Get a `VolatileSlice` from the buffer
+    /// # let vslice = mem_ref.get_slice(0, 32)
+    /// #    .expect("Could not get VolatileSlice");
```

Comment:
We should keep these lines un-hidden as well so the asserts make sense in the generated documentation.